### DISCUSSION TITLE
Localization without subpath or subdomain routing

### DISCUSSION THREAD

**Discussion by KristineTrona** [vercel/next.js#50214] [Help]
### Summary

Hello, I am looking for a way to implement localization without subdomain or subpath routing and be able to get the locale information in both server side and client side rendered components. The locale should be determined by either a query param, a cookie or by detecting the browser's language.

Currently we have implemented a middleware that depending on the query param in the request sets a cookie. Then in a server side component we get the locale information from the cookie. The disadvantage of this approach, however is that we are unable to use locale directly in client side components (unless we pass it down as a prop, which is not ideal in my opinion) and we also cannot detect browser's language in server side components. 

Anyone has any tips on how to achieve localization without specifying the locale in the url?

**Comment by bassrock** [comment]
Also looking for a solution that allows this.

**Comment by TommysG** [comment]
Take look at this article. It's a cookie based implementation for app dir without changing the path. 
[cookie translation implementation](https://carlogino.com/blog/nextjs-app-dir-i18n-cookie)
